src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Apprenticeship.cs

```diff
@@ -144,5 +152,86 @@ public DraftApprenticeship CreateCopyForChangeOfParty(ChangeOfPartyRequest chang
 
             return result;
         }
+
+        private void ValidateApprenticeshipForStop(DateTime stopDate, long accountId, ICurrentDateTime currentDate)
+        {
+            if (PaymentStatus == PaymentStatus.Completed || PaymentStatus == PaymentStatus.Withdrawn)
+            {
+                throw new DomainException(nameof(PaymentStatus), "Apprenticeship must be Active or Paused. Unable to stop apprenticeship");
+            }
+
+            if (Cohort.EmployerAccountId != accountId)
+            {
+                throw new DomainException(nameof(accountId), $"Employer {accountId} not authorised to access commitment {Cohort.Id}, expected employer {Cohort.EmployerAccountId}");
+            }
+
+            if (this.IsWaitingToStart(currentDate))
+            {
+                if (stopDate.Date != StartDate.Value.Date)
+                    throw new DomainException(nameof(stopDate), "Invalid stop date. Date should be value of start date if training has not started.");
+            }
+            else
+            {
+                /// When asking for a stop date, only a month and year are provded by the UI, The day is not supplied.
+                /// As a result, when constructing comparisons, it is clear the dates must also be of the same format.
+                if (stopDate.Date > new DateTime(currentDate.UtcNow.Year, currentDate.UtcNow.Month, 1))
+                {
+                    throw new DomainException(nameof(stopDate), "Invalid Stop Date. Stop date cannot be in the future and must be the 1st of the month.");
+                }
+
+                if (stopDate.Date < new DateTime(StartDate.Value.Year, StartDate.Value.Month, 1))
+                {
+                    throw new DomainException(nameof(stopDate), "Invalid Stop Date. Stop date cannot be before the apprenticeship has started.");
+                }
+            }
+        }
+
+        private const DataLockErrorCode CourseChangeErrors = DataLockErrorCode.Dlock03 | DataLockErrorCode.Dlock04 | DataLockErrorCode.Dlock05 | DataLockErrorCode.Dlock06;
```

Comment:
Missed this one before - there are extension methods for this (or at least, an extensions class where such methods might be added, if needs be) - `DataLockStatusExtensions.cs`

Reply:
Updated to use the correct extension 👍